packages/list_bloc/lib/src/cubits/list_cubit.dart

```diff
@@ -20,5 +20,62 @@ class ListCubit<T, F> extends DataCubit<List<T>, F> {
       rethrow;
     }
   }
+
 // TODO(alexis): implement remove, add, replace etc
+
+  Future<void> remove(int index, [F? filter]) async {
+    final f = filter ?? state.filter;
+
+    emit(Data.loading(data: state.data, filter: f));
+    try {
+      final data = [...state.data ?? []];
+      data.removeAt(index);
+      emit(
+        data.isEmpty
+            ? Data.empty(filter: f)
+            : Data(data: data, filter: state.filter),
+      );
+    } on Exception catch (e) {
+      emit(Data.error(data: state.data, filter: state.filter, error: e));
+      rethrow;
+    }
+  }
+
+  Future<void> insert(int index, [F? filter]) async {
```

Comment:
insert should insert an object at the given index in the current list of items from the state.